src/test/java/org/jenkinsci/plugins/oic/PluginTest.java

```diff
@@ -943,6 +977,52 @@ public void loginWithCheckTokenFailure() throws Exception {
         assertAnonymous();
     }
 
+    @Test
+    public void testAccessUsingJenkinsApiTokens() throws Exception {
+        mockAuthorizationRedirectsToFinishLogin();
+        configureWellKnown(null, null, "authorization_code");
+        jenkins.setSecurityRealm(new TestRealm(wireMockRule, null, EMAIL_FIELD, GROUPS_FIELD, AUTO_CONFIG_FIELD));
+        // login and assert normal auth is working
+        mockTokenReturnsIdTokenWithGroup(PluginTest::withoutRefreshToken);
+        mockUserInfoWithTestGroups();
+        browseLoginPage();
+        assertTestUser();
+
+        // create a jenkins api token for the test user
+        String token = User.getById(TEST_USER_USERNAME, false)
+                .getProperty(ApiTokenProperty.class)
+                .generateNewToken("foo")
+                .plainValue;
+
+        // validate that the token can be used
+        HttpResponse<String> rsp = getPageWithGet(TEST_USER_USERNAME, token, "/whoAmI/api/xml");
+        MatcherAssert.assertThat("response should have been 200\n" + rsp.body(), rsp.statusCode(), is(200));
```

Comment:
I assume this fails if `isTraditionalApiTokenAccessEnabled == false`,  right?

Reply:
yes, that is what [this next bit](https://github.com/jenkinsci/oic-auth-plugin/pull/386/files/9dafa6a67326e37371c4583b187de3fbc7216074#diff-fbf2979fa131a62f8efe7b73f007ba2048436733e151acf28b80836a70a7e901R1009) of the test checks